### ISSUE TITLE
Feature Request: hs.application:getContextMenuItems()/selectContextMenuItem()

### ISSUE THREAD

**Issue by dmattera** [Hammerspoon/hammerspoon#2382]
Not sure if this is even possible and/or feasible but just wanted to throw it out there just in case as I've got a few situations in my workflow where this would be useful (basically, functions in apps that are not available as MenuItems, only in certain context menus).

Also figured that since getMenuItems() can be labor intensive enough as it is, this should be a seperation HS function rather than bundled into the existing function?

**Comment by latenitefilms** [comment]
This will be certainly possible to add to Hammerspoon once [`hs.axuielement`](https://github.com/Hammerspoon/hammerspoon/pull/2373) is merged into the master.

In the meantime, you can use [`hs._asm.axuielement`](https://github.com/asmagill/hs._asm.axuielement) to achieve this.

We currently do this a lot in [CommandPost](https://github.com/CommandPost/CommandPost/) using [`hs._asm.axuielement`](https://github.com/asmagill/hs._asm.axuielement).

What would be helpful is if you could share some screenshots/examples of what exact contextual menu items you'd like to access?

CC @asmagill

**Comment by asmagill** [comment]
It *may* be possible once `hs.axuielement` is merged in... most context menus don't exist until they are showing, so there is nothing to be retrieved.  See https://github.com/asmagill/hs._asm.axuielement/blob/master/examples/dockStuff.lua#L110-L127 -- you can't look for the menu item until *after* the `doPress()` command is sent to the application button.
